src/main/io/gps.c

```diff
@@ -573,110 +848,249 @@ static void ubloxSetMessageRate(uint8_t messageClass, uint8_t messageID, uint8_t
     tx_buffer.payload.cfg_msg.msgClass = messageClass;
     tx_buffer.payload.cfg_msg.msgID = messageID;
     tx_buffer.payload.cfg_msg.rate = rate;
-    ubloxSendConfigMessage(&tx_buffer, MSG_CFG_MSG, sizeof(ubxCfgMsg_t));
+    ubloxSendConfigMessage(&tx_buffer, MSG_CFG_MSG, sizeof(ubxCfgMsg_t), false);
 }
 
-static void ubloxSetNavRate(uint16_t measRate, uint16_t navRate, uint16_t timeRef)
+static void ubloxSetMessageRateValSet(ubxValgetsetBytes_e msgClass, uint8_t rate)
 {
     ubxMessage_t tx_buffer;
 
+    uint8_t payload[1];
+    payload[0] = (uint8_t)(rate >> (8 * 0));
+
+    size_t offset = ubloxValSet(&tx_buffer, msgClass, payload, UBX_VAL_LAYER_RAM);
+
+    ubloxSendConfigMessage(&tx_buffer, MSG_CFG_VALSET, offsetof(ubxCfgValSet_t, cfgData) + offset, true);
+}
+
+static void ubloxDisableNMEAValSet(void)
+{
+    ubxMessage_t tx_buffer;
+
+    uint8_t payload[1];
+
+    payload[0] = (uint8_t)(0 >> (8 * 0));
```

Comment:
?

Reply:
Resolved in a recent commit, and similar entries, other than in larger patterns.